lib/services/sso_mfa.go

```diff
@@ -34,5 +34,11 @@ type SSOMFASessionData struct {
 	// ConnectorType is SSO type of the corresponding Auth connector (SAML, OIDC).
 	ConnectorType string `json:"connector_type,omitempty"`
 	// ChallengeExtensions are Teleport extensions that apply to this SSO MFA session.
-	ChallengeExtensions *mfav1.ChallengeExtensions `json:"challenge_extensions,omitempty"`
+	ChallengeExtensions *ChallengeExtensions `json:"challenge_extensions"`
+}
+
+// ChallengeExtensions is a json struct for [mfav1.ChallengeExtensions].
+type ChallengeExtensions struct {
```

Comment:
Could we place this type into a less generic package? Why lib/services?

Should this be somewhere under api/types, with a single shared type between lib/services and lib/auth/webauthntypes?

Reply:
I'll add `lib/auth/mfa/session.go`. We could move the `WebauthnSessionData` here too.

Reply:
Maybe in a follow up, but I wouldn't worry too much.

Reply:
It's also worth thinking about the import paths. lib/auth/webauthn pulling from lib/auth/webauthntypes is fine (the latter is a "leaf" package), but pulling from lib/auth/mfa not so much. I think I would leave it as-is (or consider a move to another leaf package, likely in api/, but as-is seems better for now).

Reply:
I'm not sure I follow, `lib/auth/mfa` is a brand new leaf package which only imports the mfav1 proto, right? Also better to keep this out of `api/` since it's purely for internal Teleport.

Reply:
Actually yeah, since this is more of a leaf pacakge than wantypes, I moved `ChallengeExtensions` out of wantypes into `lib/auth/mfa`. I'm going to move `wantypes.SessionData` too, in the final commit, we could move it to a follow up PR if it's too much.

Reply:
Moving `SSOMFASessionData` out of `lib/services` actually causes issues with /e. If we want to do this still, we can do it in a follow up PR.